compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -1815,7 +1816,7 @@ impl<'a> Linker for PtxLinker<'a> {
     }
 
     fn link_whole_rlib(&mut self, path: &Path) {
-        self.cmd.arg("--rlib").arg(path);
+        self.cmd.arg("--whole-rlib").arg(path);
```

Comment:
>I'll try to send a PR adjusting API naming in this file today.

Done in https://github.com/rust-lang/rust/pull/120099.

Reply:
In a different comment we discuss using a new linker-flavor. If that is the case we can also remove `--rlib` if there's a better way to convey the same information.

When that is said, it's quite convenient that the `llvm-bitcode-linker` knows if it's linking a `rlib` since it is not an entirely standard object archive (it includes some metadata files).

When it comes to `--whole-rlib` an even more important distinction is provided to the linker. It lets it know if the symbols needs to be kept or not. Basically symbols in `core` is not part of the `API` when compiling to a cdylib so these symbols can be internalized while symbols from "user code" needs to be kept. What is the preferred difference in syntax for letting the linker know about this?